panic when use batch put:  panic: runtime error: index out of range [69340] with length 69290

**my pebble version:crl-release-22.1**
**when i use batch put concurrently,it will panic like this :**

```
panic: runtime error: index out of range [69340] with length 69290

goroutine 23 [running]:
github.com/cockroachdb/pebble.(*Batch).prepareDeferredKeyValueRecord(0xc000129200, 0x28, 0x28, 0x1)
	C:/Go/src/github.com/cockroachdb/pebble/batch.go:458 +0x232
github.com/cockroachdb/pebble.(*Batch).SetDeferred(...)
	C:/Go/src/github.com/cockroachdb/pebble/batch.go:523
github.com/cockroachdb/pebble.(*Batch).Set(0xc000129200, {0xc000344540, 0x28, 0x0}, {0xc000344540, 0x28, 0x0}, 0x0)
	C:/Go/src/github.com/cockroachdb/pebble/batch.go:505 +0x45
dbOption/database.(*pebbleBatch).Put(0x0, {0xc000344540, 0x0, 0x0}, {0xc000344540, 0x0, 0x0})
	D:/GoWorkSpace/src/dbOption/database/database.go:426 +0x2c
main.putBatchParallel.func1(0x271)
	D:/GoWorkSpace/src/dbOption/main/main.go:149 +0x10c
created by main.putBatchParallel
	D:/GoWorkSpace/src/dbOption/main/main.go:142 +0x12c
panic: runtime error: index out of range [79041] with length 79041

goroutine 24 [running]:
github.com/cockroachdb/pebble.(*Batch).prepareDeferredKeyValueRecord(0xc000129200, 0x28, 0x28, 0x1)
	C:/Go/src/github.com/cockroachdb/pebble/batch.go:428 +0x26e
github.com/cockroachdb/pebble.(*Batch).SetDeferred(...)
	C:/Go/src/github.com/cockroachdb/pebble/batch.go:523
github.com/cockroachdb/pebble.(*Batch).Set(0xc000129200, {0xc00037f3e0, 0x28, 0x0}, {0xc00037f3e0, 0x28, 0x0}, 0x0)
	C:/Go/src/github.com/cockroachdb/pebble/batch.go:505 +0x45
dbOption/database.(*pebbleBatch).Put(0x0, {0xc00037f3e0, 0x0, 0x0}, {0xc00037f3e0, 0x0, 0x0})
	D:/GoWorkSpace/src/dbOption/database/database.go:426 +0x2c
main.putBatchParallel.func1(0x4e2)
	D:/GoWorkSpace/src/dbOption/main/main.go:149 +0x10c
created by main.putBatchParallel
	D:/GoWorkSpace/src/dbOption/main/main.go:142 +0x12c
```

**my code like this:**
```
func NewLDBDatabase(file string, cache int64, handles int) (*PbDatabase, error) {
	cacher := pebble.NewCache(cache/2 * MiB)
	defer cacher.Unref()
	opts := &pebble.Options{
		Cache:                       cacher,
		FormatMajorVersion:          pebble.FormatNewest,
		L0CompactionThreshold:       2,
		L0StopWritesThreshold:       1000,
		LBaseMaxBytes:               64 << 20, // 64 MB
		Levels:                      make([]pebble.LevelOptions, 7),
		MaxConcurrentCompactions:    8,
		MaxOpenFiles:                handles,//16384,
		MemTableSize:                64 << 20,
		MemTableStopWritesThreshold: 4,

	}
	for i := 0; i < len(opts.Levels); i++ {
		l := &opts.Levels[i]
		l.BlockSize = 32 << 10       // 32 KB
		l.IndexBlockSize = 256 << 10 // 256 KB
		l.FilterPolicy = bloom.FilterPolicy(10)
		l.FilterType = pebble.TableFilter
		if i > 0 {
			l.TargetFileSize = opts.Levels[i-1].TargetFileSize * 2
		}
		l.EnsureDefaults()
	}
	opts.Levels[6].FilterPolicy = nil
	opts.FlushSplitBytes = opts.Levels[0].TargetFileSize

	opts.EnsureDefaults()

	flag , err := PathExists(file)
	if err != nil {
		logger.Println("PathExists err:", err)
		return nil, err
	}
	if !flag {
		err := os.MkdirAll(file, os.ModePerm)
		if err != nil {
			logger.Println("os.Mkdir err:", err)
			return nil, err
		}
	}


	db, err := pebble.Open(file,opts)
	if err != nil {
		return nil, fmt.Errorf("failed to open pebble database: %w", err)
	}
	store := &PbDatabase{
		db:      db,
		closing: make(chan struct{}),
		logger: logger,
	}

	return store, nil
}
func NewDB()(database.Database,error){
	db,err := database.NewLDBDatabase("D:\\GoWorkSpace\\src\\dbOption\\data",2048/12,16384))
	if err != nil {
		logger.Fatalf("could not open database: %v", err)
		return nil, err
	}
	return db,nil
}

func main(){
	db,_ := NewDB()
	batch := db.NewBatch()
	amount := 5000
	sos := make([][]byte,amount)

	wg := &sync.WaitGroup{}

	for i:= 0;i<amount;i++{
		sos[i] = []byte(RandKey(40))
	}
	wg.Add(1)
	t := time.Now()
	go putBatchParallel(wg,sos,batch)
	wg.Wait()
	fmt.Println("putBatchParallel:",time.Since(t))
}
func RandKey(length int) string {
	str := "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
	bytes := []byte(str)
	result := []byte{}
	rand.Seed(time.Now().UnixNano()+ int64(rand.Intn(100)))
	for i := 0; i < length; i++ {
		result = append(result, bytes[rand.Intn(len(bytes))])
	}
	return string(result)
}

func putBatchParallel(wg *sync.WaitGroup, sos [][]byte, db database.Batch) {
	defer wg.Done()
	defer db.Destroy()

	mod := len(sos) / runtime.NumCPU() 
	if mod < 30 {                      
		mod = 30
	}
	var selfWg sync.WaitGroup
	selfWg.Add(len(sos)+mod)
	for i := 0; i < len(sos)+mod; i += mod { 
		go func(i int) {
			defer selfWg.Done()
			var value,key []byte
			for j := i; j < (i+mod) && j < len(sos); j++ { 
				key = sos[j]
				value = sos[j]
				err := db.Put(key,value)
				if err != nil {
					panic(err)
				}
			}
		}(i)
	}
	selfWg.Wait()
	db.Write()
}
```

--- a/batch.go
+++ b/batch.go
@@ -78,7 +78,9 @@ func (d DeferredBatchOp) Finish() error {
 // RangeKeyUnsets, and/or RangeKeyDeletes that are applied atomically. Batch
 // implements the Reader interface, but only an indexed batch supports reading
 // (without error) via Get or NewIter. A non-indexed batch will return
-// ErrNotIndexed when read from .
+// ErrNotIndexed when read from. A batch is not safe for concurrent use, and
+// consumers should use a batch per goroutine or provide their own
+// synchronization.
 //
 // Indexing
 //

